Move Weights and Fees changes and updates to the new repo (#168)
substrate-developer-hub/substrate-docs: v3/docs/03-runtime/h-weights-and-fees/index.mdx

}
```

## Next steps

> The entire logic of fees is encapsulated in `pallet-transaction-payment` via a `SignedExtension`.

Reviewer: ```suggestion
The entire logic of fees is encapsulated in `pallet-transaction-payment` via a [`SignedExtension`](/concepts/extrinsics#signed-extension).
```

> While this module provides a high degree of flexibility, a user can opt to build their custom

Reviewer: ```suggestion
While this pallet provides a high degree of flexibility, a user can opt to build their custom
```

payment module drawing inspiration from Transaction Payment.

> Given now you know what Substrate weight system is, how it affect the transaction fee computation, and how to specify 

Reviewer: ```suggestion
Given now you know what Substrate's weight system is, how it affects transaction fee computation, and how to specify 
```

them for your dispatchables, the last question is how to find the right weights for your dispatchables. 
That is what **Substrate benchmarking** is for. By writing benchmarking functions and running them, the system 
(`frame-benchmarking`) calls these functions repeatedly with different numerical parameters and empirically determine 
the weight functions for dispatchables in their worst case scenarios, within a certain limit. For more information, see [Benchmarking](../i-benchmarking).

- Dedicated [weight documentation](/v3/concepts/weight)
- [Example module](https://github.com/paritytech/substrate/blob/master/frame/example/src/lib.rs)
- [SignedExtension](/rustdocs/latest/sp_runtime/traits/trait.SignedExtension.html)

### Examples

You can find examples of custom weights and fees in the following repositories:

> * [Custom weights](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/weights)

Reviewer: ```suggestion
* [Custom weights for the Example pallet](https://github.com/paritytech/substrate/blob/master/frame/example/src/weights.rs)
```

> * [custom WeightToFee](https://github.com/substrate-developer-hub/recipes/tree/master/runtimes/weight-fee-runtime).

Reviewer: ```suggestion
* [Custom WeightToFee how-to guide](/how-to-guides/v3/weights/calculate-fees)
```
